### PR TITLE
kernel: bump 5.15 to 5.15.49

### PR DESCRIPTION
**Pull request by rsalvaterra**
Deleted (upstreamed):
generic/backport-5.15/400-mtdblock-warn-if-opened-on-NAND.patch [01]
generic/backport-5.15/420-v5.19-01-mtd-spinand-gigadevice-fix-Quad-IO-for-GD5F1GQ5UExxG.patch [02]
generic/backport-5.15/702-v5.19-34-net-ethernet-mtk_eth_soc-fix-misuse-of-mem-alloc-int.patch [03]
bcm27xx/patches-5.15/950-0029-Revert-mailbox-avoid-timer-start-from-callback.patch [04]
bcm27xx/patches-5.15/950-0417-bcm2711_thermal-Don-t-clamp-temperature-at-zero.patch [05]
bcm27xx/patches-5.15/950-0740-drm-vc4-hvs-Fix-frame-count-register-readout.patch [06]
bcm27xx/patches-5.15/950-0755-drm-vc4-hvs-Reset-muxes-at-probe-time.patch [07]
bcm27xx/patches-5.15/950-0759-drm-vc4-txp-Don-t-set-TXP_VSTART_AT_EOF.patch [08]
bcm27xx/patches-5.15/950-0760-drm-vc4-txp-Force-alpha-to-be-0xff-if-it-s-disabled.patch [09]
bcm53xx/patches-5.15/031-v5.17-0005-ARM-dts-BCM5301X-update-CRU-block-description.patch [10]
mediatek/patches-5.15/102-mt7629-enable-arch-timer.patch [11]

Manually rebased:
bcm27xx/patches-5.15/950-0741-drm-vc4-hvs-Use-pointer-to-HVS-in-HVS_READ-and-HVS_W.patch

[01] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=f41c9418c5898c01634675150696da290fb86796
[02] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=d5b66645305c6f3a1b2cf75cee4157b07f293309
[03] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.49&id=fcba12a4308143f1f1dcb0face8b3ffb51edcc5c
[04] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=119f99209d8531359bcb935f252ec435f9d21b13
[05] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=c4e1280abead1552e1764684079a43e222ccd163
[06] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=08465a1889cb48ec64431e9db745b5be15399251
[07] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=0e26a6da02e63b75b629573d13966c36aa6264f0
[08] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=01c9020b2e7c85e394879f34851805179ac3d1d8
[09] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=d47f85cc0171a5d3c5bd8cbb8a98983ca3357cbd
[10] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=5151f24ac937ff7eb1f078257c66e3c0f0296010
[11] https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=ffea838686b82fbb2801cdfad6ba5309d15c032d

Signed-off-by: Rui Salvaterra \<rsalvaterra@gmail.com\>

### CODE REVIEW

File: target/linux/mediatek/patches-5.15/102-mt7629-enable-arch-timer.patch [hunk 1]
```diff
@@ -1,10 +1,10 @@
 --- a/arch/arm/mach-mediatek/Kconfig
 +++ b/arch/arm/mach-mediatek/Kconfig
```
Comment (981213) [inline target/linux/mediatek/patches-5.15/102-mt7629-enable-arch-timer.patch:2]: This patch is upstreamed:

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.46&id=ffea838686b82fbb2801cdfad6ba5309d15c032d

The auto-refresh applies this to mt7623 and...

File: target/linux/mediatek/patches-5.15/103-mt7623-enable-arch-timer.patch [hunk 2]
```diff
@@ -10,11 +10,11 @@ Signed-off-by: Chuanhong Guo <gch981213@gmail.com>
 
 --- a/arch/arm/mach-mediatek/Kconfig
 +++ b/arch/arm/mach-mediatek/Kconfig
-@@ -26,6 +26,7 @@ config MACH_MT6592
- config MACH_MT7623
- 	bool "MediaTek MT7623 SoCs support"
+@@ -22,6 +22,7 @@ config MACH_MT6589
+ config MACH_MT6592
+ 	bool "MediaTek MT6592 SoCs support"
```
Comment (981213) [inline target/linux/mediatek/patches-5.15/103-mt7623-enable-arch-timer.patch:15]: ...the actual mt7623 patch changes different lines.

File: target/linux/mediatek/patches-5.15/102-mt7629-enable-arch-timer.patch (hunk 1 above)
Comment (rsalvaterra) [reply target/linux/mediatek/patches-5.15/102-mt7629-enable-arch-timer.patch:2]: Ugh… quilt trying to be smarter than it should, I guess. Will fix…

File: target/linux/mediatek/patches-5.15/103-mt7623-enable-arch-timer.patch (hunk 2 above)
Comment (rsalvaterra) [reply target/linux/mediatek/patches-5.15/103-mt7623-enable-arch-timer.patch]: And done too.

**Comment by graysky2** [comment]
```
Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200
```

**Comment by graysky2** [comment]
```
Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200
```

**Comment by rsalvaterra** [comment]
> 5.15.48 released

Oh, cute, more broken speculation mitigations. Sigh. How depressing…

**Comment by graysky2** [comment]
5.15.48
```
Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200
```

**Comment by pkgadd** [comment]
This needs https://patchwork.ozlabs.org/project/openwrt/patch/20220610082003.2548-1-rsalvaterra@gmail.com/, even with the default gcc(-11) for x86, with that config addition applied on top:

```
kver: 5.15.48
Build system: x86_64
Build-tested: x86_64
Run-tested: x86_64 (ivy-bridge/ c1037u)
```

**Comment by ldir-EDB0** [comment]
@rsalvaterra Don't know if https://git.openwrt.org/?p=openwrt/staging/ldir.git;a=shortlog;h=refs/heads/kbump is of any help to you?

**Comment by rsalvaterra** [comment]
Rebased on top of current master.

> This needs https://patchwork.ozlabs.org/project/openwrt/patch/20220610082003.2548-1-rsalvaterra@gmail.com/

Patch added to the pull request.

> @rsalvaterra Don't know if https://git.openwrt.org/?p=openwrt/staging/ldir.git;a=shortlog;h=refs/heads/kbump is of any help to you?

I haven't really checked, but it's probably nice to have another reference.

**Comment by graysky2** [comment]
5.14.49
```
Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200
```

**Comment by graysky2** [comment]
5.14.50 is an automatic bump: https://gist.github.com/graysky2/94414b2d44e0cff124dbc3257675e0f3
```
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200
```

**Comment by hauke** [comment]
Thank you for your patches, I added them to the master branch in 251a382c28854ea68bd46483a58990eafbf671ca, b55efebb1fe99bae953adb8bd7743a61ffc40fb9, f2795132bae896374e21f45b3ea75396c9de8b34 and f1cd14448221d6114c6c150a8e78fa360bbb47dd.

I removed the generic/backport-5.15/702-v5.19-32-net-ethernet-mtk_eth_soc-out-of-bounds-read-in-mtk_h.patch patch, it is already upstream:    
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.47&id=b24ca1cf846273361d5bd73a35de95a486a54b6d

**Pull request closed without merging**